Handle older channels.http.AsgiRequest that uses a different attribute name for request.environ

cb109 opened the pull request:
The channels.http.AsgiRequest uses `request.__dict__["META"]` instead of `request.__dict__["environ"]`. This gave me a `KeyError` when using the ASGI/Channels version 2.4.0 development server (aka `runserver` when `settings.ASGI_APPLICATION` is configured).

I don't know if you are interested in supporting this setup, but since it was such a small fix I though I'd mention it. 
Tested with Python 3.7.9, Django 3.2.14, channels 2.4.0 on Ubuntu 20.04.

Thanks for sharing the method, I am quite excited about these code stacks telling me where all those queries come from!

cb109 committed:
Handle older AsgiRequest VS WsgiRequest using different attribute name

The channels.http.AsgiRequest uses request.META instead of
request.environ, at least for the somewhat older v2.4.0 it does.

Tested with Python 3.7.9 and Django 3.2.14 on Ubuntu 20.04.

contextref commented on req_stats/middleware.py:
```diff
@@ -223,7 +223,12 @@ def do_logging(
+        except KeyError:
+            # Handle v2.x channels.http.AsgiRequest using a different key:
+            # https://github.com/django/channels/blob/2.4.0/channels/http.py#L61
+            resp_env = request.__dict__["META"]  # type: dict
```
Thanks for suggesting this fix.  Brief research and testing tells me that `META` is in fact always there in current Django versions, so we can just switch to that unconditionally -- that's the approved way per the doc.  Would you mind switching to META without the condition?  Or I can make that change -- whatever you think is best.

cb109 replied:
Oh I wasn't aware of that, thanks for investigating, pushed!

cb109 committed:
Use META key which is available on both WSGI/ASGI requests

In response to https://github.com/Path-AI/django-request-stats-example/pull/3#discussion_r1051645649
